### ISSUE TITLE
Clickhouse can't start if table with very long name is in database

### ISSUE THREAD

**Issue by miv** [ClickHouse/ClickHouse#36485]
**Describe the unexpected behaviour**

If a very long table name is present in any database, **clickhouse can't start** 

_**This table can't be dropped**_

**How to reproduce**
* Which ClickHouse server version to use

22.3

Create a table with very long name (as in example below).

Try to drop it - fails
Try to restart clickhouse - fails

**Expected behavior**

Dropping tables should work as expected

**Error message and/or stacktrace**

`2022.04.21 07:41:52.011372 [ 1 ] {} <Error> Application: filesystem error: in posix_stat: failed to determine attributes for the specified path: File name too long [/var/lib/clickhouse/metadata/kw/temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_temp_kw_bi_testkey.sql.detached]`

**Additional context**

The cause is trivial - too long file name. Clickhouse renames some files with longer filenames ('.detached') and because name was already at maximum length it can't do it.

Even if cause is trivial, this is not trivial to fix. Should clickhouse check if file can be saved with all suffixes it can add? Should it somehow know max length of file on mounted filesystem (it can not)? Should it rename files to `<sha>.<suffix>` to always be less than 83 symbols (maximum length in FAT)?

**Comment by den-crane** [comment]
and if you use Cyrillic to Japanese characters or Chinese characters it even worse.

```
create table "куку"("хаха" Int64) Engine=MergeTree order by tuple();
insert into "куку" select * from numbers(100000000);
/var/lib/clickhouse/data/default/%D0%BA%D1%83%D0%BA%D1%83/all_85_90_1/%D1%85%D0%B0%D1%85%D0%B0.bin
```

**Comment by movy** [comment]
Still happening in  24.1.2.5

```
Error: std::exception. Code: 1001, type: std::__1::__fs::filesystem::filesystem_error, e.what() = filesystem error: in rename: File name too long ["/var/lib/clickhouse/store/6e3/6e39a927-1c11-414c-aef4-416fc143c698/_tmp_market_data_2024_03_14_volume__entry_alerts_manual_exp51_ent_alrt_man_2024_03_13T11_34_37_lastPNL__4_symsCount_0_sharpe_1_wr_0_6_avghrs_10_POSITIVE_RT_ONLY_LVavgPnl_0_4_pck_01e1a7d919b3bcaccf20c13df7449e63.sql"] ["/var/lib/clickhouse/metadata_dropped/default._tmp_market_data_2024_03_14_volume__entry_alerts_manual_exp51_ent_alrt_man_2024_03_13T11_34_37_lastPNL__4_symsCount_0_sharpe_1_wr_0_6_avghrs_10_POSITIVE_RT_ONLY_LVavgPnl_0_4_pck_01e1a7d919b3bcaccf20c13df7449e63.29e58cd6-855e-4f38-843f-39099f0035b2.sql"]

```

**Comment by bonifacemwanza** [comment]
was this fixed in the new clickhouse version ?

**Comment by al13n321** [comment]
Yes, should be fixed by https://github.com/ClickHouse/ClickHouse/pull/85326
